engines/instance_verification/spec/requests/api/connect/v3/systems/products_controller_spec.rb

```diff
@@ -20,200 +21,587 @@
 
-        it 'reports an error' do
-          data = JSON.parse(response.body)
-          expect(data['error']).to eq('Instance verification failed: The product is not available for this instance')
+        context 'when a suitable subscription is found' do
+          let(:product) do
+            FactoryBot.create(
+              :product, :with_mirrored_repositories, :extension, free: false, base_products: [base_product]
+              )
+          end
+          let(:product_classes) { [base_product.product_class, product.product_class] }
+          let(:plugin_double) { instance_double('InstanceVerification::Providers::Example') }
+
+          it 'returns error when SCC call fails' do
+            data = JSON.parse(response.body)
+            expect(data['error']).to eq('Instance verification failed: The product is not available for this instance')
+          end
         end
       end
+    end
+
+    context 'when activating extensions without errors' do
+      let(:instance_data) { 'dummy_instance_data' }
+      let(:system) do
+        FactoryBot.create(
+          :system, :hybrid, :with_system_information, :with_activated_product, product: base_product, instance_data: instance_data
+          )
+      end
+      let(:serialized_service_json) do
+        V3::ServiceSerializer.new(
+          product.service,
+          base_url: URI::HTTP.build({ scheme: response.request.scheme, host: response.request.host }).to_s
+          ).to_json
+      end
+      let(:scc_activate_url) { 'https://scc.suse.com/connect/systems/products' }
+      let(:plugin_double) { instance_double('InstanceVerification::Providers::Example') }
+      let(:payload) do
+        {
+          identifier: product.identifier,
+          version: product.version,
+          arch: product.arch,
+          instance_data: 'dummy_instance_data',
+          proxy_byos_mode: :hybrid,
+          hwinfo:
+          {
+            hostname: 'test',
+            cpus: '1',
+            sockets: '1',
+            hypervisor: 'Xen',
+            arch: 'x86_64',
+            uuid: 'ec235f7d-b435-e27d-86c6-c8fef3180a01',
+            cloud_provider: 'amazon'
+          }
+        }
+      end
+
+      before do
+        allow(InstanceVerification::Providers::Example).to receive(:new)
+          .with(nil, nil, nil, instance_data).and_return(plugin_double)
+        allow(plugin_double).to receive(:parse_instance_data).and_return({ InstanceId: 'foo' })
+
+        FactoryBot.create(:subscription, product_classes: product_classes)
+        stub_request(:post, scc_activate_url)
+          .to_return(
+            status: 201,
+            body: {}.to_json,
+            headers: {}
+          )
+        # stub the fake announcement call PAYG has to do to SCC
+        # to create the system before activate product (and skip validation)
+        stub_request(:post, 'https://scc.suse.com/connect/subscriptions/systems')
+          .to_return(status: 201, body: { ok: 'OK' }.to_json, headers: {})
```

Comment:
The response from SCC is a hash with the system's login + password (docs: https://scc.suse.com/connect/v4/documentation#/subscriptions/post_subscriptions_systems): 

```
{
  "id": 1234567,
  "login": "SCC_3b336b126db1503a9513a14e92a6a62e",
  "password": "24f057b7941e80f9cf2d51e16e8af2d6"
}
```

In case of the HYBRID announcement call, RMT needs to include the login/password in the announcement payload. It's already implemented in SCC to allow announcement with preset login/pw.

Reply:
what about the response body for `POST connect/systems/products` ?